pkg/jobs/job_info_storage.go

```diff
@@ -138,19 +146,32 @@ func (i InfoStorage) write(ctx context.Context, infoKey, value []byte) error {
 }
 
 func (i InfoStorage) iterate(
-	ctx context.Context, infoPrefix []byte, fn func(infoKey, value []byte) error,
+	ctx context.Context,
+	iterMode iterateMode,
+	infoPrefix []byte,
+	fn func(infoKey, value []byte) error,
 ) (retErr error) {
 	if i.txn == nil {
 		return errors.New("cannot iterate over the job info table without an associated txn")
 	}
 
+	var iterConfig string
+	switch iterMode {
+	case iterateAll:
+		iterConfig = `ORDER BY info_key ASC, written DESC` // with no LIMIT.
```

Comment:
given that for a job_id there will only ever be one row per `info_key` we could drop the secondary sort on written, right?